pkg/scale/encode.go

```diff
@@ -284,22 +382,46 @@ func (es *encodeState) encodeFixedWidthInt(i interface{}) (err error) {
 	switch i := i.(type) {
 	case int8:
 		err = binary.Write(es, binary.LittleEndian, byte(i))
+		if err != nil {
+			err = fmt.Errorf("encoding FixedWidthInt: %w", err)
```

Comment:
shouldn't this and all the other fixed width cases use a sentinel error as well? `ErrEncodeFixedWidthInt`